tests/kernel/mem_heap/k_heap_api/src/test_kheap_api.c

```diff
@@ -40,6 +40,40 @@ static void thread_alloc_heap(void *p1, void *p2, void *p3)
 
 /*test cases*/
 
+/* These need to be adjacent in BSS */
+volatile uint32_t heap_guard0;
+K_HEAP_DEFINE(tiny_heap, 1);
+volatile uint32_t heap_guard1;
```

Comment:
Hm... can toolchain relocate them freely, so these won't be together?

Reply:
They will on existing platforms.  You're right that with -fdata-sections you could write a linker script to break this.  But in practice linker directives that select symbols with the same wildcard preserve their order within the file, and it's no doubt to preserve behavior in code exactly like this.

And in any case, the side effect would be that the weird platform with a pathological linker script gets a false "pass" result on this particular test case in the face of an overrun that gets caught on more conventional platforms.

I'm comfortable with it, anyway :)